Source/DataModels/Components/ComponentAnimation.cpp

```diff
@@ -107,12 +107,20 @@ void ComponentAnimation::Update()
 				static_cast<ComponentTransform*>(owner->GetComponent(ComponentType::TRANSFORM))->UpdateTransformMatrices();
 			}
 		}
+		lastState = actualState;
+	}
+	else
+	{
+		lastState == NON_STATE;
```

Comment:
This is most likely wrong
```suggestion
		lastState = NON_STATE;
```